daml/Marketplace/Lifecycle/Model.daml

```diff
@@ -95,18 +100,27 @@ template Effect
     bondRegistrar : Party
     cashProvider : Party
     lifecycleId : Text
+    cfi: CFI
     assetId : Id
     newVersion : Int
-    date : Date
+    recordDate: Date
+    payDate : Date
     payouts : [(Decimal, Id)]
+    remainingClaims : Claims
     expired : Bool
+    isin: Text
   where
     signatory operator, issuer, bondRegistrar
 
+    key (bondRegistrar, lifecycleId): (Party, Text)
+    maintainer key._1
+
     nonconsuming choice RequestSnapshot : ContractId SnapshotRequest
       controller bondRegistrar
       do
-        create SnapshotRequest with custodian = bondRegistrar; ..
+        (_, description) <- fetchByKey @AssetDescription.T assetId
+        create description with assetId.version = newVersion; claims = remainingClaims
```

Comment:
Any particular reason why the creation of the new AssetDescription moved from the CalculateLifecycleEffect choice to here?